examples/mining_cfd.py

```diff
@@ -0,0 +1,127 @@
+import desbordante
+import pandas
+
+# CFD Discovery parameters:
+TABLEPATH = 'examples/datasets/university_fd.csv'
+MINIMUM_SUPPORT = 8
+MINIMUM_CONFIDENCE = 0.7
+MAXIMUM_LHS_COUNT = 3
+
+
+GREEN_BG_CODE = '\033[1;42m'
+RED_BG_CODE = '\033[1;41m'
+DEFAULT_BG_CODE = '\033[1;49m'
+
+
+def items_to_indices(items):
+    return [item.attribute for item in items]
+
+
+def find_var_attributes(items):
+    result = []
+    for item in items:
+        if item.value is None:
+            result.append(item.attribute)
+    return result
+
+
+def row_is_supported(row_dataframe, pattern):
+    for item in pattern:
+        if item.value is not None and str(row_dataframe.iloc[0, item.attribute]).lower() != item.value.lower():
+            return False
+    return True
+
+
+# returns a dictionary
+#   key: a string representation of a dataframe with certain attribute values
+#   value: indices of all rows with those attribute values
+def make_lhs_to_row_nums(cfd_lhs_var_attributes, supported_rows, table):
+    lhs_to_row_nums = {}
+    for row_num in range(0, len(table.index)):
+        row_dataframe = table.iloc[[row_num], :]
+        if row_num not in supported_rows:
+            continue
+        if row_dataframe.iloc[0, cfd_lhs_var_attributes].to_string() in lhs_to_row_nums:
+            lhs_to_row_nums[row_dataframe.iloc[0, cfd_lhs_var_attributes].to_string()].append(row_num)
+        else:
+            lhs_to_row_nums[row_dataframe.iloc[0, cfd_lhs_var_attributes].to_string()] = [row_num]
+    return lhs_to_row_nums
+
+
+def get_supported(lhs, table):
+    supported = []
+    for row_num in range(0, len(table.index)):
+        row_dataframe = table.iloc[[row_num], :]
+        if row_is_supported(row_dataframe, lhs):
+            supported.append(row_num)
+            continue
+    return supported
+
+
+def validate_cfd(lhs, rhs, table):
+    cfd_lhs_var_attributes = find_var_attributes(lhs)
+    supported_rows = get_supported(lhs, table)
+    rows_satisfying_cfd = []
+    lhs_to_row_nums = make_lhs_to_row_nums(cfd_lhs_var_attributes, supported_rows, table)
+
+    # mark rows with identical lhs with green color if they have the most frequent rhs
+    for string_lhs in lhs_to_row_nums:
```

Comment:
Use [dict.items](https://docs.python.org/3.8/library/stdtypes.html#dict.items), same for rhs